### DISCUSSION TITLE
useDrag for overlapping meshes in R3F React Three Fiber

### DISCUSSION THREAD

**Discussion by btjacker** [pmndrs/use-gesture#578] [Q&A] [answered]
Hello and thank you for this awesome library. 
I have started to learn react recently so maybe is trivial question. I have modified public codesandbox and added another object to different position. If you rotate the camera so that both objects are overlapping and then drag one of them, they both will move. Is there a way how to invoke the event only for the closest object in R3F? Thank you for help.
[codesandbox](https://codesandbox.io/s/affectionate-beaver-kszkkg)

**Comment by btjacker** [comment, marked as answer]
Offcourse event.stopPropagation(); is the solution. Sorry for making mess with useless questions

**Reply by dbismut** [reply]
Haha no worries. Join the Discord if you want quicker answers.

**Comment by aksong-propane** [comment]
The only way I've been able to get stopPropagation to work is to remove the delay: true configuration option, but this has unintended consequences.  Is there a way to have both work?

Here is my fork of the [codesandbox](https://codesandbox.io/p/sandbox/nice-curie-z6z8ss?workspaceId=fc0f01b9-0da8-4115-ac56-5af0073a482f&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clyi9lut20006356jcklz2dtg%2522%252C%2522sizes%2522%253A%255B100%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clyi9lut20002356jvv6tn065%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clyi9lut20003356jufpayypx%2522%257D%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clyi9lut20005356j5fv2cdfg%2522%257D%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clyi9lut20002356jvv6tn065%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clyi9lut20001356jtx6yf9ft%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.jsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clyi9lut20002356jvv6tn065%2522%252C%2522activeTabId%2522%253A%2522clyi9lut20001356jtx6yf9ft%2522%257D%252C%2522clyi9lut20005356j5fv2cdfg%2522%253A%257B%2522id%2522%253A%2522clyi9lut20005356j5fv2cdfg%2522%252C%2522activeTabId%2522%253A%2522clyicjbkb000l356j25d65ho9%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clyi9lut20004356j8bymsgip%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522clyi9lx7v000p356jimfxidyx%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522clyicjbkb000l356j25d65ho9%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522clyi9lut20003356jufpayypx%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clyi9lut20003356jufpayypx%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)
